ducktape/cluster/cluster.py

```diff
@@ -38,7 +46,19 @@ def __len__(self):
         """Size of this cluster object. I.e. number of 'nodes' in the cluster."""
         raise NotImplementedError()
 
-    def request(self, nslots):
+    def request_subcluster(self, num_nodes):
```

Comment:
I was following the `request` convention we established for getting nodes.

These methods are only really used internally, so it wouldn't have impact to change both `request` and `request_subcluster` to `alloc` and `alloc_cluster` for example